android/src/main/java/com/audiowaveform/AudioWaveformModule.kt

```diff
@@ -191,19 +192,24 @@ class AudioWaveformModule(context: ReactApplicationContext): ReactContextBaseJav
 
     @ReactMethod
     fun seekToPlayer(obj: ReadableMap, promise: Promise) {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            val progress = obj.getInt(Constants.progress)
-            val key = obj.getString(Constants.playerKey)
-            if (key != null) {
-                audioPlayers[key]?.seekToPosition(progress.toLong(), promise)
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                val progress = obj.getInt(Constants.progress)
+                val key = obj.getString(Constants.playerKey)
+                if (key != null) {
+                    audioPlayers[key]?.seekToPosition(progress.toLong(), promise)
```

Comment:
Is there any reason to keep forwarding the promise inside the player and pollute the audio player class?
Would it not be safer and cleaner to keep promises inside the module that is doing the bridge between native code and JS?